--- a/src/nft_ext_fullcone.c
+++ b/src/nft_ext_fullcone.c
@@ -410,7 +410,7 @@ static const struct nft_expr_ops nft_fullcone_ipv6_ops = {
        .size           = nft_fullcone_size,
        .eval           = nft_fullcone_eval,
        .init           = nft_fullcone_init,
-       .dump           = nft_fullcone_dump,
+       .dump           = nft_fullcone_dump,
        .validate       = nft_fullcone_validate,
 };
